simphony/cuds/material.py

```diff
@@ -0,0 +1,31 @@
+import uuid
+
+from simphony.core.data_container import DataContainer
+
+
+class Material(object):
+    """A class that represents a material.
+
+    Class describes a material and its data.
+
+    Attributes
+    ----------
+    description : str
+        textual description of the material
+    uid : uuid.UUID
+        the id of the material. If None, then it is generated
+    data : DataContainer
+        data associated with this material
+
+    """
+    def __init__(self, uid=None, data=None, description=""):
```

Comment:
>  If I am not wrong (and I may be ;-)), this is used in the serialization process and the link with other objects, no?

@dpinte  yes, that is correct.  

We use the unique id notion mostly (e.g. `"single_particle = particles.get_particle(particle.uid)`) but we do divert from using uid notion for the high-level datasets as they have a string `name` that is used to identify them (e.g. `foo_dataset = engine.get_dataset("foo")`).